### ISSUE TITLE
How to use the Rivescript.deparse() method

### ISSUE THREAD

**Issue by arashsa** [aichaos/rivescript-python#76]
From the source code:

```
Dump the in-memory RiveScript brain as a Python data structure.

This would be useful, for example, to develop a user interface for
editing RiveScript replies without having to edit the RiveScript
source code directly.
```

I've just started looking into the source code of the project, and I'm trying to get my head around this method. I see that it parses everything to a python object, but how can you edit replies, and then load it back into Rivescript again?

**Comment by kirsle** [comment]
The usage of these commands is a little clunky. :wink: I originally implemented it for the Perl version (back when it was the *only* version) for a web based chatbot editor tool, so its usage was geared around that use case:

1. Use `deparse()` to get a Python data structure that you can edit.
2. Use `write(filename, deparsed)` to write your data structure as a RiveScript source file.
3. Use `load_file()` in the future to load that generated source file.

Full example:

```python
deparsed = bot.deparse()
deparsed["topic"]["random"]["hello bot"] = {
    "reply": ["Hello human."],
}
bot.write("custom.rive", deparsed)

# ...
bot.load_file("custom.rive")
```

To make the functions a little more useful, the module can be updated to allow "reversing" the `deparse()` function; for example, if you *give* `deparse()` an equivalent data structure it can put it into the bot's brain.

```python
def deparse(self, deparsed=None):
    if type(deparsed) is dict:
        # loop through it and put things into the bot's memory...
    else:
        # the current logic of exporting the bot's memory and returning it
```

**Comment by arashsa** [comment]
I tried calling the deparse method, but I get the error:

```
Traceback (most recent call last):
  File "bot_rest_api_rs.py", line 24, in <module>
    deparsed = rs.deparse()
  File "/Users/arashsaidi/.pyenv/versions/ml/lib/python3.5/site-packages/rivescript/rivescript.py", line 336, in deparse
    for trig, data in self._topics[topic].iteritems():
AttributeError: 'list' object has no attribute 'iteritems'
```

**Comment by kirsle** [comment]
> `AttributeError: 'list' object has no attribute 'iteritems'`

I see this function was never updated to be Python 2 & 3 compatible. It should use the cross-version-compatible `.items()` instead of `.iteritems()` which was Python 2-only.

I'll put it on my To-Do list to fix the bug so that this function works in Python3, and while I'm in there I'll implement the "reverse deparse" I mentioned earlier.

**Comment by arashsa** [comment]
Sounds good! :)

**Comment by smuthusa** [comment]
Kindly let me know when we can expect the fix?

**Comment by kirsle** [comment]
This is fixed now in v1.14.8
